cowprotocol/services
# Use quote id just for debugging

# Description
Passing the quote id during the order creation request has currently weird and not well documented behavior.
Essentially it's used to streamline fetching quotes from the DB and allows us to better link an order creation request with a quote request.
If you provide no quote id the backend computes a fresh quote in the order creation requests so this usually never fails. However, if you provide a quote id we are currently throwing errors if the referenced quote is expired or the parameters don't match.
That means passing the **optional** quote id (that the openapi scheme documents just as a debugging tool) can only be detrimental for the user compared to passing no id at all.

# Changes
This PR makes it such that this ID is actually only used for debugging and can not introduce new error variants. That means if we can find a quote based on the order's search parameters we just use that but if we can't find it we always try to compute a fresh one. This eliminates 2 error cases (quote expired, or parameter mismatch) and makes it easier to reason about the code.

## How to test
the (unit) test coverage on that logic was already pretty good so I mostly had to adjust some tests (to please the compiler) and remove a few test cases for the 2 removed error variants.

1 e2e test needed to be updated. The test relied on using the quote id to make sure the backend looks up an outdated quote while creating the order.
So previously we could:
1. get a quote
2. update liquidity sources
3. create order with outdated quote id causing it to get flagged as limit / market order depending on the quote

But now we have to make sure the order would get classified as market / limit at the time of creation. Before we could use the liquidity sources as a synchronization mechanism but now that's no longer possible. As an easy (and more explicit) way around that I added a function to conveniently enable or disable a solver account to make sure no solutions get settled until we are ready.
